runtime/src/iree/hal/drivers/vulkan/vulkan_device.cc

```diff
@@ -1257,6 +1300,26 @@ static iree_status_t iree_hal_vulkan_device_query_i64(
     return iree_ok_status();
   }
 
+  if (iree_string_view_equal(
+          category,
+          iree_make_cstring_view("hal.device.vulkan.subgroup_operations"))) {
+    uint32_t requested_subgroup_operations = 0;
+    if (!iree_string_view_atoi_uint32(key, &requested_subgroup_operations)) {
```

Comment:
instead of doing the test in here we would want to just return the value and test on the outside - so this should be a "hal.dispatch" query of "subgroup_operations"/"subgroup_features" that returns an i64 bitfield. this way we have one single query that gets cached vs one query per unique set of bits a user is testing.